rust-semantics/rust-common-syntax.md

```diff
@@ -257,8 +257,7 @@ https://doc.rust-lang.org/reference/items/extern-crates.html
                               | ByteLiteral | ByteStringLiteral | RawByteStringLiteral
                               | CStringLiteral | RawCStringLiteral
                               | IntegerLiteral | FloatLiteral
-                              | "true" | "false"
```

Comment:
Are you sure these aren't needed here? Would this grammar still parse a .rs file containing true/false? Even if the file is still parsable (e.g. it may identify true/false as identifiers), why is it a good idea to remove these?

Reply:
My motivation for removing them was to primarily eliminate parsing ambiguity in the execution module. Having a syntax element named `true` of type `Expression` was causing issues on, for instance, the `check_eq true` in the testing files, as it didn't recognize if this `true` was an `Expression` or a `Bool`. 

Keeping `Bool` as a part of `Values` is also advantageous for us as we do not need to write new rules for assignments of constants and variables, and also makes it easier for us to reuse the return of the relational operations of `MInt`s.

Finally, please, correct me if I'm wrong, but my understanding is that an `Expression` can also be constituted by a single `Value`. So having Booleans as a part of `Values` allows us to have effectively the same syntax of adding `true` and `false` to the `Expression` definition, with the advantage of having the boolean operations implemented in the BOOL module being supported here.